### PR TITLE
assorted fixes

### PR DESCRIPTION
**Pull request by janisozaur**
just some fixes

### CODE REVIEW

File: src/ride/track.c [hunk 1]
```diff
@@ -370,7 +370,7 @@ void track_list_populate(ride_list_item item, uint8* track_list_cache){
 		}
 
 		// If cur_track_entry_index is greater than max number of tracks
-		if (cur_track_entry_index >= 1000){
+		if (cur_track_entry_index >= 254){
```
Comment (IntelOrca) [inline src/ride/track.c]: Where did you get 254 from, and why have you changed this?

Comment (janisozaur) [inline src/ride/track.c]: 1000 seems to be a arbitrary watch value, however, `cur_track_entry_index` is of type `uint8` (line 341) thus 254 is the highest sane value with which we can compare here.

Comment (IntelOrca) [inline src/ride/track.c]: Something is wrong then, the original code couldn't have been comparing a byte with a word. This routine will need checking and cleaning up. 1000 is the maximum number of tracks allocated in memory though for a ride's track design list...

Comment (janisozaur) [inline src/ride/track.c]: Other than this, does the rest of this commit look good for merging? I can revert this line and create a separate issue for it.

Comment (IntelOrca) [inline src/ride/track.c]: I think so... the peep direction looks like it could be a severe bug so that's a good find.

**Comment by janisozaur** [comment]
I updated the commit as discussed

**Pull request merged**